server/plugin/hooks.go

```diff
@@ -47,14 +54,59 @@ func (p *Plugin) MessageHasBeenPosted(c *plugin.Context, post *model.Post) {
 		return
 	}
 
+	if len(post.FileIds) > 1 {
+		p.logAndSendErrorToUser(mattermostUserID, channel.Id, "Cannot send more than one file attachment at a time.")
+		return
+	}
+
 	token, err := p.ParseAuthToken(user.OAuth2Token)
 	if err != nil {
 		p.logAndSendErrorToUser(mattermostUserID, channel.Id, fmt.Sprintf("error occurred while decrypting token. Error: %s", err.Error()))
 		return
 	}
 
+	var attachment *MessageAttachment
+	if len(post.FileIds) == 1 {
+		fileInfo, appErr := p.API.GetFileInfo(post.FileIds[0])
+		if appErr != nil {
+			p.logAndSendErrorToUser(mattermostUserID, channel.Id, fmt.Sprintf("Error getting file info. Error: %s", appErr.Message))
+			return
+		}
+
+		date := time.Now()
+		//TODO: Add a configuration setting for expiry time
+		expiryTime := date.Add(time.Minute * 15)
+
+		file := &FileStruct{
+			ID:     post.FileIds[0],
+			Expiry: expiryTime,
+		}
+
+		var jsonBytes []byte
+		jsonBytes, err = json.Marshal(file)
+		if err != nil {
+			p.API.LogError("Error occurred while mashaling file. Error: %s", err.Error())
+		}
+
+		var encrypted []byte
+		encrypted, err = encrypt(jsonBytes, []byte(p.getConfiguration().EncryptionSecret))
+		if err != nil {
+			p.API.LogError("Error occurred while encrpting file. Error: %s", err.Error())
+		}
+
+		encoded := encode(encrypted)
+
+		fileLink := p.GetPluginURL() + "/file/" + encoded
+
+		attachment = &MessageAttachment{
+			URL:         fileLink,
```

Comment:
can we use its value directly here instead of a variable? same for `encoded` as well